templates/nextflow-aurora-mysql.yaml

```diff
@@ -125,6 +152,137 @@ Resources:
+  BootstrapLambdaTrigger:
+    Condition: UseSnapshotFalse
+    Type: Custom::BootstrapLambdaTrigger
+    Properties:
+      ServiceToken: !GetAtt BootstrapLambdaFunction.Arn
+
+  BootstrapLambdaRole:
+    Condition: UseSnapshotFalse
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+              - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: "/"
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole'
+      Policies:
+        -
+          PolicyName: "SecretAccess"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              -
+                Effect: "Allow"
+                Action: "secretsmanager:GetSecretValue"
+                Resource:
+                  - !Ref AuroraMasterSecret
+                  - !Ref AuroraTowerUserSecret
+        -
+          PolicyName: "DataAccess"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              -
+                Effect: "Allow"
+                Action: "rds-data:*"
+                Resource: "*"
+
+  BootstrapLambdaFunction:
+    Condition: UseSnapshotFalse
+    Type: AWS::Lambda::Function
+    Properties:
+      Description: Bootstrap newly Created Aurora Database
+      Handler: index.lambda_handler
+      Role: !GetAtt BootstrapLambdaRole.Arn
+      Runtime: python3.7
+      Environment:
+        Variables:
+          Cluster_ARN: !Sub 'arn:aws:rds:${AWS::Region}:${AWS::AccountId}:cluster:${AuroraCluster}'
+          Master_Secret_ARN: !Ref AuroraMasterSecret
+          Tower_User_Secret_ARN: !Ref AuroraTowerUserSecret
+          Region: !Ref 'AWS::Region'
+          DBName: !Ref DBName
+      Code:
+        ZipFile: |
+          import boto3
+          import cfnresponse
+          import logging
+          from os import environ
+
+          log = logging.getLogger(__name__)
+          log.setLevel(logging.DEBUG)
+
+          def submit_sql(sql):
+            return boto3.client('rds-data').execute_statement(
+              secretArn=environ['Master_Secret_ARN'],
+              database=environ['DBName'],
+              resourceArn=environ['Cluster_ARN'],
+              schema='mysql',
+              sql=sql
+            )
+
+
+          def bootstrap():
+            try:
+              secret_response = boto3.client('secretsmanager').get_secret_value(SecretId=environ['Tower_User_Secret_ARN'])
+              secret_string = secret_response['SecretString']
+              alter_db_response = submit_sql("ALTER DATABASE tower CHARACTER SET utf8 COLLATE utf8_bin;")
+              log.debug(f"Alter DB response: {alter_db_response}")
+              create_user_response = submit_sql("CREATE USER 'tower' IDENTIFIED BY '" + secret_string + "';")
```

Comment:
per above suggestion, instead of the static tower db username maybe you can parameterize it and allow passing in any username?

Reply:
sure